contracts/0.8.9/oracle/AccountingOracle.sol

```diff
@@ -700,174 +710,145 @@ contract AccountingOracle is BaseOracle {
+            dataOffset = iter.dataOffset;
+        }
+    }
 
-        uint256 itemType;
+    function _processExtraDataItem(bytes calldata data, ExtraDataIterState memory iter) internal {
+        uint256 dataOffset = iter.dataOffset;
         uint256 moduleId;
-        uint256 lastNodeOpId;
-
-        while (i < items.length) {
-            (uint256 iIndex, uint256 iType, uint216 iPayload) = _decodeExtraDataItem(items[i]);
+        uint256 nodeOpsCount;
+        uint256 firstNodeOpId;
+        bytes calldata nodeOpIds;
+        bytes calldata valsCounts;
+
+        if (dataOffset + 35 > data.length) {
+            // has to fit at least moduleId (3 bytes), nodeOpsCount (8 bytes),
+            // and data for one node operator (8 + 16 bytes), total 35 bytes
+            revert InvalidExtraDataItem(iter.index);
+        }
 
-            if (iIndex != iter.firstItemIndex + i) {
-                revert UnexpectedExtraDataIndex(iter.firstItemIndex + i, iIndex);
-            }
+        /// @solidity memory-safe-assembly
+        assembly {
+            // layout at the dataOffset:
+            // | 3 bytes  |   8 bytes    |  nodeOpsCount * 8 bytes  |  nodeOpsCount * 16 bytes  |
+            // | moduleId | nodeOpsCount |      nodeOperatorIds     |      validatorsCounts     |
+            let header := calldataload(add(data.offset, dataOffset))
+            moduleId := shr(232, header)
+            nodeOpsCount := and(shr(168, header), 0xffffffffffffffff)
+            nodeOpIds.offset := add(data.offset, add(dataOffset, 11))
+            nodeOpIds.length := mul(nodeOpsCount, 8)
+            firstNodeOpId := shr(192, calldataload(nodeOpIds.offset))
+            valsCounts.offset := add(nodeOpIds.offset, nodeOpIds.length)
+            valsCounts.length := mul(nodeOpsCount, 16)
+            dataOffset := sub(add(valsCounts.offset, valsCounts.length), data.offset)
+        }
 
-            (uint256 iModuleId, uint256 iNodeOpId, uint256 iKeysCount) =
-                _decodeExtraDataPayload(iPayload);
+        if (moduleId == 0) {
+            revert InvalidExtraDataItem(iter.index);
+        }
 
-            if (started) {
-                if (iType != itemType || iModuleId != moduleId) {
-                    break;
-                }
-                if (iNodeOpId <= lastNodeOpId) {
-                    revert InvalidExtraDataSortOrder();
-                }
-            } else {
-                if (int256(iType) < iter.lastType || int256(iType) == iter.lastType && (
-                    int256(iModuleId) < iter.lastModuleId ||
-                    int256(iModuleId) == iter.lastModuleId && int256(iNodeOpId) <= iter.lastNodeOpId
-                )) {
-                    revert InvalidExtraDataSortOrder();
-                }
-                itemType = iType;
-                moduleId = iModuleId;
-                started = true;
+        unchecked {
+            // | 2 bytes  | 41 bytes | 3 bytes  |    8 bytes    |
```

Comment:
41 -> 19

Reply:
have no idea how did I arrive at 41, def need to get more sleep